<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>settingDialog</class>
 <widget class="QDialog" name="settingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>418</width>
    <height>316</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QListWidget" name="listWidget">
     <property name="minimumSize">
      <size>
       <width>400</width>
       <height>70</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>70</height>
      </size>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="dragDropMode">
      <enum>QAbstractItemView::NoDragDrop</enum>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="textElideMode">
      <enum>Qt::ElideLeft</enum>
     </property>
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="viewMode">
      <enum>QListView::IconMode</enum>
     </property>
     <property name="currentRow">
      <number>-1</number>
     </property>
     <item>
      <property name="text">
       <string>General</string>
      </property>
      <property name="textAlignment">
       <set>AlignLeft|AlignVCenter</set>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>:/im/images/setting.png</normaloff>:/im/images/setting.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Computer</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>:/im/images/computer.png</normaloff>:/im/images/computer.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Hub</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>:/im/images/hub.png</normaloff>:/im/images/hub.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Switch</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>:/im/images/switch.png</normaloff>:/im/images/switch.png</iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Router</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>:/im/images/router.png</normaloff>:/im/images/router.png</iconset>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>150</height>
      </size>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="general">
      <layout class="QVBoxLayout" name="verticalLayout_13">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Lifetime arp-entries: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="sp_ttlArp">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="suffix">
            <string> s</string>
           </property>
           <property name="minimum">
            <number>10</number>
           </property>
           <property name="maximum">
            <number>1200</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Animation speed:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="sl_speed">
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksBothSides</enum>
           </property>
           <property name="tickInterval">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>38</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="computer">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Nubmer of interfaces by default:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="computerComboBox">
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>70</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="hub">
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Number of ports by default</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="hubComboBox">
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>12</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>24</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>48</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>70</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="switch_2">
      <layout class="QVBoxLayout" name="verticalLayout_10">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Number of ports by default: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="switchComboBox">
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>12</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>16</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>24</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>32</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>48</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_9">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Lifetime records of table switching</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="sp_ttlMac">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="suffix">
            <string> s</string>
           </property>
           <property name="minimum">
            <number>10</number>
           </property>
           <property name="maximum">
            <number>300</number>
           </property>
           <property name="value">
            <number>300</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="router">
      <layout class="QVBoxLayout" name="verticalLayout_12">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_11">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Number of ports by default: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="routerComboBox">
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>70</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btn_apply">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Apply</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/im/images/refresh.png</normaloff>:/im/images/refresh.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_ok">
       <property name="text">
        <string>Ok</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/im/images/ok.png</normaloff>:/im/images/ok.png</iconset>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_cancel">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/im/images/not.png</normaloff>:/im/images/not.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>btn_ok</tabstop>
  <tabstop>btn_apply</tabstop>
  <tabstop>btn_cancel</tabstop>
  <tabstop>sp_ttlArp</tabstop>
  <tabstop>computerComboBox</tabstop>
  <tabstop>hubComboBox</tabstop>
  <tabstop>switchComboBox</tabstop>
  <tabstop>sp_ttlMac</tabstop>
  <tabstop>routerComboBox</tabstop>
  <tabstop>listWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>listWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>97</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_cancel</sender>
   <signal>clicked()</signal>
   <receiver>settingDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>591</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>637</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_ok</sender>
   <signal>clicked()</signal>
   <receiver>settingDialog</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>computerComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>258</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_apply</sender>
   <signal>clicked()</signal>
   <receiver>settingDialog</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>403</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sp_ttlArp</sender>
   <signal>valueChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hubComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>37</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>switchComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sp_ttlMac</sender>
   <signal>valueChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>186</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_apply</sender>
   <signal>clicked()</signal>
   <receiver>settingDialog</receiver>
   <slot>applyDisable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>routerComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>settingDialog</receiver>
   <slot>applyEnable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>105</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>apply()</slot>
  <slot>applyEnable()</slot>
  <slot>applyDisable()</slot>
 </slots>
</ui>
